Aplib.Core/Desire/Goals/Goal.cs

```diff
@@ -28,14 +28,9 @@ public virtual Heuristics CurrentHeuristics(BeliefSet beliefSet)
             => _currentHeuristics ??= _heuristicFunction.Invoke(beliefSet);
 
         /// <summary>
-        /// The name used to display the current goal during debugging, logging, or general overviews.
+        /// Data about the Goal such as a name and description, this may be useful for debugging or logging.
```

Comment:
XML comments for public (auto) properties should start with 'Gets' (And 'Gets or sets' if set is also an option).

Also, try to summarize the property in one sentece. If you want to add extra information, add it in a <remark> tag:

```suggestion
        /// Gets or sets the metadata of the goal.
```

You can set the details in a remark:
```
/// <remark>
/// The metadata may be useful for debugging
/// </remark>
```

(For some reason, I could not set a suggestion for that, sorry.

Reply:
This is also the case for Action and Tactic. Please edit those accordingly.